sickbeard/webserve.py

```diff
@@ -3643,6 +3643,21 @@ def saveGeneral(self, log_dir=None, log_nr = 5, log_size = 1048576, web_port=Non
         sickbeard.DOWNLOAD_URL = download_url
         sickbeard.INDEXER_DEFAULT_LANGUAGE = indexerDefaultLang
         sickbeard.LAUNCH_BROWSER = config.checkbox_to_value(launch_browser)
+        if sickbeard.SHOWUPDATE_HOUR != config.to_int(showupdate_hour):
+            sickbeard.showUpdateScheduler.stop.set()
+            logger.log(u"Waiting for the SHOWUPDATER thread to exit so we can set new start hour")
+            try:
+                sickbeard.showUpdateScheduler.join(10) # Wait 10 sec for the thread to exit
+            except:
+                pass
+            if  sickbeard.showUpdateScheduler.isAlive():
+                logger.log(u"Unable to stop SHOWUPDATER thread, the new configuration will be applied after a restart", logger.WARNING)
+            else:
+                logger.log(u"Starting SHOWUPDATER thread with the new start hour: " + str(showupdate_hour))
+                sickbeard.showUpdateScheduler = scheduler.Scheduler(showUpdater.ShowUpdater(),
+                                              cycleTime=datetime.timedelta(hours=1),
+                                              threadName="SHOWUPDATER",
+                                              start_time=datetime.time(hour=int(showupdate_hour)))            
```

Comment:
You should use `config.to_int(showupdate_hour)` instead of `int(showupdate_hour)` to expect the same value behavior.